### DISCUSSION TITLE
Accessing host from container

### DISCUSSION THREAD

**Discussion by g0loob** [testcontainers/testcontainers-dotnet#447] [Q&A] [answered]
Is there a recommended (cross-platform) way how to access host from a container? On Windows OS it works with `host.docker.internal`, but I don't know how to access on Linux (e.g. GitLab CI/CD pipeline).

---

My use case is that I have [Pact](https://pact.io)'s mock server running on host machine. It can either listen on `127.0.0.1` or `0.0.0.0`. In the test I need to start a Docker container and pass to it mock server's address as a command argument, e.g.

```shell
docker container run --rm my_image:tag -u mock-server.url # other arguments
```

I'm aware that I might need to replace the mock server with a Docker container, but I'm interested if there's a way with the current setup.

**Comment by HofmeisterAn** [comment, marked as answer]
I think for Linux, there is no default DNS name. I don't know much about the GitLab environment, but if you have a running container, you should get the host address from the `Hostname` property:

https://github.com/HofmeisterAn/dotnet-testcontainers/blob/de4c4a251152ee2cfdcd4061c8b1bcc0621e692f/src/DotNet.Testcontainers/Containers/TestcontainersContainer.cs#L85

If your service listens on `0.0.0.0`, you should be able to access it. The Kafka configuration, for instance, uses the property during the container start to configure it own services:

https://github.com/HofmeisterAn/dotnet-testcontainers/blob/de4c4a251152ee2cfdcd4061c8b1bcc0621e692f/src/DotNet.Testcontainers/Configurations/Modules/MessageBrokers/KafkaTestcontainerConfiguration.cs#L78

**Reply by g0loob** [reply]
Thanks for the response. I'll mark it as answer, it will probably work in a "normal" scenario. I forgot that my tests in GitLab are executed in a container. I used [container's ID](https://stackoverflow.com/a/49406631/4778343) to inspect it and get its IP address. Using the IP address of a container in which the tests are executed, and which hosts the mock server, I was able to access the mock server from the other container.
